src/bench/addrman.cpp

```diff
@@ -70,13 +70,22 @@ static void FillAddrMan(CAddrMan& addrman)
 
 static void AddrManAdd(benchmark::Bench& bench)
 {
-    CreateAddresses();
+     /* Create many CAddrMan objects - one to be modified at each loop iteration. */
 
-    CAddrMan addrman(/* deterministic */ false, /* consistency_check_ratio */ 0);
+    bench.epochs(5).epochIterations(1);
```

Comment:
Could also remove the make_unique and use the beautiful curly brackets?

```cpp
static void AddrManAdd(benchmark::Bench& bench)
{
    CreateAddresses();

    bench.run([&] { 
        CAddrMan addrman{/* deterministic */ false, /* consistency_check_ratio */ 0};
        AddAddressesToAddrMan(addrman);
    }); 
}

Reply:
The reason to construct the addrman objects outside the bench run is to prevent the time spent in constructing the object being included in the bench. I think that would be especially bad and could lead to variance if we're allocating memory for the object by making a unique_ptr.

I've reverted to Marco's suggested version, which shouldn't be any worse than the current code, which calls `Clear()`.